pkg/apimachinery/patch/patch.go

```diff
@@ -38,6 +38,91 @@ const (
 	PatchRemoveOp  = "remove"
 )
 
+func (p *PatchOperation) MarshalJSON() ([]byte, error) {
+	switch p.Op {
+	// The 'remove' operation is the only patching operation without a value
+	// and it needs to be parsed differently.
+	case PatchRemoveOp:
+		return json.Marshal(&struct {
+			Op   string `json:"op"`
+			Path string `json:"path"`
+		}{
+			Op:   p.Op,
+			Path: p.Path,
+		})
+	case PatchTestOp, PatchReplaceOp, PatchAddOp:
+		return json.Marshal(&struct {
+			Op    string      `json:"op"`
+			Path  string      `json:"path"`
+			Value interface{} `json:"value"`
+		}{
+			Op:    p.Op,
+			Path:  p.Path,
+			Value: p.Value,
+		})
+	default:
+		return nil, fmt.Errorf("operation %s not recognized", p.Op)
+	}
+}
+
+type PatchSet struct {
+	patches []PatchOperation
+}
+
+type PatchOption func(patches *PatchSet)
+
+func New(opts ...PatchOption) *PatchSet {
+	p := &PatchSet{}
+	p.AddOption(opts...)
+	return p
+}
+
+func (p *PatchSet) AddOption(opts ...PatchOption) {
+	for _, f := range opts {
+		f(p)
+	}
+}
+
+func (p *PatchSet) addOp(op, path string, value interface{}) {
```

Comment:
nit. I think it's clearer this way
```suggestion
func (p *PatchSet) addPatchOperation(patchOperation, path string, value interface{}) {
```

Reply:
I find the current name used is elegant and to the point.

I think a common reader understand this small method name and can decode it as "add operation". Making the name longer is not really helpful. It is also internal, with even less incentive to expand the naming.

Reply:
Yes, agree with Ed. First, it is a local function and then the entire package is already dealing with patch operation, hence, it is a bit redundant

Reply:
Personally I don't really understand the value of having short names. Longer and clearer names (up to a reasonable limit of course) are pretty much always better since there's no trade-off here - only a win as nowadays IDEs can easily auto-complete names.

In any case this is just a nit.